presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -211,9 +212,14 @@ public Principal getUserPrincipal()
         return Optional.empty();
     }
 
-    private static void sendChallenge(HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
+    private static void sendChallenge(HttpServletRequest request, HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
         throws IOException
     {
+        // skip request body before sending the status to avoid dropping of the connection
+        // if connection is dropped before all the content is sent by jetty client, the client request will be marked as failed,
```

Comment:
This text is a bit long for a comment and makes it harder to read. I'd wrap it at ~100 characters.